### PR TITLE
Make TimescaleDB available with JDBC syntax

### LINKED ISSUE #3890: Make TimescaleDB available with JDBC syntax
As a Developer i want to use TimescaleDB in Testcontainers similar to PostGIS.

Currently a PostGIS container can be started when following JDBC Url is given: 
`jdbc:tc:postgis:9.6-2.5:///databasename`

I want to have the same behavior for Timescale DB. I propose following JDBC Url: 
`jdbc:tc:timescaledb:///databasename`

### PR DESCRIPTION
**Pull request by raynigon**
Fix  #3890

The Syntax for the use of a TimescaleDB Testcontainer should be `jdbc:tc:timescaledb:///databasename` or `jdbc:tc:timescaledb:2.1.0-pg13:///databasename` if a Tag is given. This should allow the user to use the container without any further configuration.

# Checklist
## Default docker image

- [x] Should be a Docker Hub official image, or published by a reputable source (ideally the company or organisation that officially supports the technology)
- [x] Should have a verifiable open source Dockerfile and a way to view the history of changes
- [x] MUST show general good practices regarding container image tagging - e.g. we do not use `latest` tags, and we do not use tags that may be mutated in the future
- [x] MUST be legal for Testcontainers developers and Testcontainers users to pull and use. Mechanisms exist to allow EULA acceptance to be signalled, but images that can be used without a licence are greatly preferred.

## Module dependencies

- [x] The module should use as few dependencies as possible, 
- [x] Regarding libraries, either:
    - they should be `compileOnly` if they are likely to already be on the classpath for users' tests (e.g. client libraries or drivers) 
    - they can be `implementation` (and thus transitive dependencies) if they are very unlikely to conflict with users' dependencies.
- [x] If client libraries are used to test or use the module, these MUST be legal for Testcontainers developers and Testcontainers users to download and use.

## API (e.g. `MyModuleContainer` class)

- [x] Favour doing the right thing, and least surprising thing, by default
- [x] Ensure that method and parameter names are easy to understand. Many users will ignore documentation, so IDE-based substitutes (autocompletion and Javadocs) should be intuitive. 
- [x] The module's public API should only handle standard JDK data types and MUST not expose data types that come from `compileOnly` dependencies. This is to reduce the risk of compatibility problems with future versions of third party libraries.
 
## Documentation

- [x] Every module MUST have a dedicated documentation page containing:
    - [x] A high level overview
    - [x] A usage example
    - [x] If appropriate, basic API documentation or further usage guidelines
    - [x] Dependency information
    - [x] Acknowledgements, if appropriate
- [x] Consider that many users will not read the documentation pages - even if the first person to add it to a project does, people reading/updating the code in the future may not. Try and avoid the need for critical knowledge that is only present in documentation.

### CODE REVIEW

File: modules/jdbc-test/src/main/java/org/testcontainers/jdbc/AbstractJDBCDriverTest.java [hunk 1]
```diff
@@ -117,7 +117,9 @@ private void performTestForJDBCParamUsage(HikariDataSource dataSource) throws SQ
         String databaseQuery = "SELECT DATABASE()";
         // Postgres does not have Database() as a function
         String databaseType = ConnectionUrl.newInstance(jdbcUrl).getDatabaseType();
-        if (databaseType.equalsIgnoreCase("postgresql") || databaseType.equalsIgnoreCase("postgis")) {
+        if (databaseType.equalsIgnoreCase("postgresql") ||
+            databaseType.equalsIgnoreCase("postgis") ||
+            databaseType.equalsIgnoreCase("timescaledb")) {
             databaseQuery = "SELECT CURRENT_DATABASE()";
         }
```
Comment (raynigon) [inline modules/jdbc-test/src/main/java/org/testcontainers/jdbc/AbstractJDBCDriverTest.java:124]: This is pretty ugly and should be made more generic, but that would be a different PR.

**Review by rnorth — APPROVED** [review]
I think I'm happy with this - it follows the pattern used for PostGIS already (although has more by way of tests, which is good).

One general topic, which I think we should consider outside of this PR, is recommendations for usage of PostGIS/TimescaleDB outside of a JDBC URL usage mode. I know it's as simple as creating a `PostgreSQLContainer` with a different image, but this isn't particularly discoverable or documented.

**Review by kiview — APPROVED** [review]
LGTM, thanks for your contribution @raynigon.

@rnorth 
I was thinking the same thing with regards to the similar Postgis scenario as well.
